internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -394,6 +396,57 @@ func (r *ClusterDeploymentsReconciler) installDay1(ctx context.Context, log logr
 	return r.updateStatus(ctx, log, clusterInstall, cluster, nil)
 }
 
+func (r *ClusterDeploymentsReconciler) spokeKubeClient(ctx context.Context, clusterDeployment *hivev1.ClusterDeployment) (spoke_k8s_client.SpokeK8sClient, error) {
+	adminKubeConfigSecretName := getClusterDeploymentAdminKubeConfigSecretName(clusterDeployment)
+
+	namespacedName := types.NamespacedName{
+		Namespace: clusterDeployment.Namespace,
+		Name:      adminKubeConfigSecretName,
+	}
+
+	secret, err := getSecret(ctx, r.Client, r.APIReader, namespacedName)
+	if err != nil {
+		r.Log.WithError(err).Errorf("failed to get kubeconfig secret %s", namespacedName)
+		return nil, err
+	}
+	if err = ensureSecretIsLabelled(ctx, r.Client, secret, namespacedName); err != nil {
+		r.Log.WithError(err).Errorf("failed to label kubeconfig secret %s", namespacedName)
+		return nil, err
+	}
+	return r.SpokeK8sClientFactory.CreateFromSecret(secret)
+}
+
+func (r *ClusterDeploymentsReconciler) updateWorkerMcpPaused(ctx context.Context, log logrus.FieldLogger, clusterInstall *hiveext.AgentClusterInstall, clusterDeployment *hivev1.ClusterDeployment) error {
```

Comment:
What is triggering the cluster deployment reconcile in this case? day2 cluster don't have any changes on day2 installation.

Reply:
Agent modifications trigger cluster deployment reconcile